configure.ac

```diff
@@ -535,20 +535,14 @@ if test "x${GCC}" = "xyes" ; then
         GCC_VER="`${CC} -dumpversion`"
         GCC_MAJOR_VER="`echo ${GCC_VER} | cut -d. -f1`"
         GCC_MINOR_VER="`echo ${GCC_VER} | cut -d. -f2`"
-        if test "${GCC_MAJOR_VER}" -ge "4" ; then
-                if test "${GCC_MINOR_VER}" -ge "6" ; then
-                        WNO_FLAGS=""
-                        AC_MSG_RESULT([>= 4.6])
-                else
-                        WNO_FLAGS=""
-                        AC_MSG_RESULT([< 4.6])
-                fi
-        else
-                AC_MSG_RESULT([< 4])
+        if test "${GCC_MAJOR_VER}" -ge "7" ; then
+            C_SUPPRESS_FLAGS=""
+            CXX_SUPPRESS_FLAGS="-Wno-deprecated"
         fi
+        AC_MSG_RESULT([${GCC_VER}])
         SHARED_FLAGS="-Wall -Wextra -Werror -Wundef -ffunction-sections -fdata-sections -Wsequence-point -pipe -Winit-self -Wmissing-include-dirs"
-        CFLAGS="$SHARED_FLAGS $CFLAGS $WNO_FLAGS"
-        CXXFLAGS="$SHARED_FLAGS -Wshadow $CXXFLAGS"
+        CFLAGS="$SHARED_FLAGS $CFLAGS $C_SUPPRESS_FLAGS"
+        CXXFLAGS="$SHARED_FLAGS -Wshadow $CXXFLAGS $CXX_SUPPRESS_FLAGS"
```

Comment:
would you consider this instead?
in a/src/vma/sock/sock-redirect.h
```
+#if __GNUC__ >= 7
+#define VMA_THROW(x)
+#else
+#define VMA_THROW(x) throw(x)
+#endif
int (*setsockopt) (int __fd, int __level, int __optname, __const void *__optval, socklen_t __optlen) VMA_THROW (vma_error);
```

this is more accurate.

Reply:
after thinking about it and reading about this how about removing the entire throw?